src/integrations/windowsIntegrationManager.ts

```diff
@@ -1,14 +1,433 @@
+    try {
+      await fs.promises.access(cliPath);
+      // Nothing to do if the file exists
+    } catch (err: any) {
+      if (err.code !== 'ENOENT') {
+        console.error(`Can't create the cli-plugins directory:`, err);
+
+        return;
+      }
+      await fs.promises.mkdir(cliDir, { recursive: true });
+
+      try {
+        await fs.promises.copyFile(srcPath, cliPath, fs.constants.COPYFILE_EXCL);
+      } catch (err2) {
+        console.error(`Failed to copy file ${ srcPath } to ${ cliPath }`, err2);
+      }
+    }
+  }
+
+  protected async syncDistroDockerCompose(distro: string) {
+    const srcPath = await this.getLinuxToolPath(distro, 'bin', 'docker-compose');
+    const destDir = '$HOME/.docker/cli-plugins';
+    const destPath = `${ destDir }/docker-compose`;
+    const state = this.settings.kubernetes?.WSLIntegrations?.[distro] === true;
+
+    console.debug(`Syncing ${ distro } docker compose: ${ srcPath } -> ${ destDir }`);
+    // Update only the distro -- the current
+    if (state) {
+      await this.execCommand({ distro }, '/bin/sh', '-c', `mkdir -p "${ destDir }"`);
+      await this.execCommand({ distro }, '/bin/sh', '-c', `if [ ! -e "${ destPath }" -a ! -L "${ destPath }" ] ; then ln -s "${ srcPath }" "${ destPath }" ; fi`);
+    } else {
+      try {
+        // This is preferred to doing the readlink and rm in one long /bin/sh statement because
+        // then we rely on the distro's readlink supporting the -n option. Gnu/linux readlink supports -f,
+        // On macOS the -f means something else (not that we're likely to see macos WSLs).
+        const targetPath = (await this.captureCommand({ distro }, '/bin/sh', '-c', `readlink -f "${ destPath }"`)).trimEnd();
+
+        if (targetPath === srcPath) {
+          await this.execCommand({ distro }, '/bin/sh', '-c', `rm "${ destPath }"`);
+        }
+      } catch (err) {
+        console.log(`Failed to readlink/rm ${ destPath }`, err);
+      }
+    }
+  }
+
+  protected async syncKubeconfig() {
+    const kubeconfigPath = await K3sHelper.findKubeConfigToUpdate('rancher-desktop');
+
+    await Promise.all((await this.distros).map(distro => this.syncDistroKubeconfig(distro, kubeconfigPath)));
+  }
+
+  protected async syncDistroKubeconfig(distro: string, kubeconfigPath: string) {
+    const state = this.settings.kubernetes?.WSLIntegrations?.[distro] === true;
+
+    try {
+      console.debug(`Syncing ${ distro } kubeconfig`);
+      if (this.settings.kubernetes?.enabled) {
+        await this.execCommand(
+          {
+            distro,
+            env: {
+              ...process.env,
+              KUBECONFIG: kubeconfigPath,
+              WSLENV:     `${ process.env.WSLENV }:KUBECONFIG/up`,
+            }
+          },
+          await this.getLinuxToolPath(distro, 'wsl-helper'),
+          'kubeconfig',
+          `--enable=${ state }`,
+        );
+      }
+    } catch (error) {
+      const errorAny = error as any;
```

Comment:
Can you not just do `catch (error:any) {`? That's what I've been doing and everything seems to work.